docs/public/components/using-components.md

```diff
@@ -0,0 +1,170 @@
+    func Commands(globalParams *command.GlobalParams) []*cobra.Command {
+        cmd := &cobra.Command{
+            Use:   "test",
+            Short: "a test command for the Agent",
+            Long:  ``,
+            RunE: func(_ *cobra.Command, _ []string) error {
+                return fxutil.OneShot(
+                    <callback>,
+                    <list of dependencies>.
+                )
+            },
+        }
+
+        return []*cobra.Command{cmd}
+    }
+    ```
+
+The code above creates a test command that does nothing. As you can see, `fxutil.OneShot` helpers are being used. These
+helpers initialize an Fx app with all the wanted dependencies. 
+
+The next section explains how to request a
+dependency.
+
+## Importing components
+
+The `fxutil.OneShot` takes a list of components and gives them to Fx. Note that this only tells Fx how to create types when they're needed. This does not do anything else.
+
+For a component to be instantiated, it must be one of the following:
+
++ Required as a parameter by the `callback` function
++ Required as a dependency from other components already marked for instantiation
++ Directly asked for by using `fx.Invoke`. More on this on the [Fx page](fx.md).
+
+Let's require the `log` components:
+
+```go
+import (
+    // First let's import the FX wrapper to require it
+    logfx "github.com/DataDog/datadog-agent/comp/core/log/fx"
+    // Then the logger interface to use it
+    log "github.com/DataDog/datadog-agent/comp/core/log/def"
+)
+
+// [...]
+    return fxutil.OneShot(
+        myTestCallback, // The function to call from fxutil.OneShot
+        logfx.Module(), // This will tell FX how to create the `log.Component`
+    )
+// [...]
+
+func myTestCallback(logger log.Component) {
+    logger.Info("some message")
+}
+```
+
+## Importing bundles
+
+Now let's say you want to include the core bundle instead. The core bundle offers many basic features (logger, config,
+telemetry, flare, ...).
+
+```go
+import (
+    // We import the core bundle
+    core "github.com/DataDog/datadog-agent/comp/core"
+
+    // Then the interfaces we want to use
+    config "github.com/DataDog/datadog-agent/comp/core/config/def"
+)
+
+// [...]
+    return fxutil.OneShot(
+        myTestCallback, // The function to call from fxutil.OneShot
+        core.Bundle(),  // This will tell FX how to create the all the components included in the bundle
+    )
+// [...]
+
+func myTestCallback(conf config.Component) {
+    api_key := conf.GetString("api_key")
+
+    // [...]
+}
+```
+
+It's very important to understand that since `myTestCallback` only uses the `config.Component`, not all components from
+the `core` bundle are instantiated! The `core.Bundle` instructs Fx how to create components, but only the ones required
+are created.
+
+In our example, the `config.Component` might have dozens of dependencies instantiated from the core bundle. Fx handles
+all of this.
+
+## Using non-component types with Fx
```

Comment:
I've usually seen this being used to supply Params structs and similar plain-old-data objects. Things that shouldn't become components because they don't define an interface of behaviors.

Perhaps this heading should be `## Using plain data types with Fx`

Reply:
Indeed, I think using `Params` is a good usage but this part seems to refer to code like [this one](https://github.com/DataDog/datadog-agent/blob/27630a86a09e6ea0e07ff5d0fe5edc3ec1e3ea27/cmd/agent/subcommands/run/command.go#L423-L425).